### PR TITLE
Add Cache option for Flows webhook trigger

### LINKED ISSUE #18271: Unable to bypass cache for webhook GET requests
### Describe the Bug

GET requests to `/flows/trigger/*` are cached, resulting in a cached response and the flow not being triggered. Query string appended to the URL does not bypass the cache.

Update: Not considered a bug as the bypassing of cache can be achieved with a `Cache-Control: no-store` header when the `CACHE_SKIP_ALLOWED` environment variable is `true`.

### To Reproduce

Create a new flow with a webhook trigger.
Ensure that `CACHE_ENABLED` is set to `true`.
Try triggering the webhook multiple times and notice it only triggering once.

### Directus Version

v9.25.1

### Hosting Strategy

Self-Hosted (Docker Image)

### PR DESCRIPTION
**Pull request by licitdev**
Add `Cache` option so that caching can be disabled for GET requests to Flows webhook trigger.

https://user-images.githubusercontent.com/26413686/234685302-11bc0a64-e319-4c36-b04d-4994349d4d4d.mov

Fixes #18271, Closes ENG-912

### CODE REVIEW

File: api/src/flows.ts [hunk 1]
```diff
@@ -235,6 +240,7 @@ class FlowManager {
 				flow.options['return'] = flow.options['return'] ?? '$last';
 
 				this.webhookFlowHandlers[`${method}-${flow.id}`] = handler;
+				this.webhookFlowPreventCaches[`${method}-${flow.id}`] = !!flow.options['preventCache'];
```
Comment (azrikahar) [inline api/src/flows.ts:243]: Currently the app does only show the option when it's a GET webhook, but I believe we'll need to check it in the api/flows side as well, specifically for this pretty rare edge case of: 


1. Create a GET webhook trigger with `preventCache` enabled, so now `flow.options['preventCache']` is true, then save.
2. Update the same webhook trigger from GET to POST. Although the option itself disappears, `flow.options['preventCache']` still has value in the db, and this will still evaluate as true.

(Though understandably "lingering" options is technically something present even for display options and interface options if I'm not mistaken, which leads to why I recall such edge case here)

**Review by azrikahar — COMMENTED** [review]
Works like a charm, especially after using a similar manual test locally like the blackbox test, using Run Script + Date to verify it'll prevent the caching.

Couple of additional thoughts:

- Let's update the docs for Webhook trigger as well:

    https://github.com/directus/directus/blob/1f7cc5e86b5eb3c5dc6d4d4b7bc3115da545b4f3/docs/app/flows/triggers.md?plain=1#L99-L126

- thoughts on `Disable Cache` vs `Prevent Cache`, perhaps at least in terms of it's name in the app? Personally don't really mind `Prevent Cache` at all, but wonder which of them would be a tad clearer for non-technical users. But this may also be cleared up by the docs in the first point!

File: api/src/controllers/flows.ts [hunk 2]
```diff
@@ -33,6 +33,10 @@ const webhookFlowHandler = asyncHandler(async (req, res, next) => {
 		}
 	);
 
+	if (preventCache) {
```
Comment (rijkvanzanten) [inline api/src/controllers/flows.ts:36]: Lets "Flip" the variables here. Lets make the vars and options "Cache Enabled" with "true" being the default, and then check against `if (cacheEnabled === false)` 👍🏻

**Review by rijkvanzanten — DISMISSED** [review]

File: api/src/controllers/flows.ts (hunk 2 above)
Comment (licitdev) [reply api/src/controllers/flows.ts]: Great idea! Done~ 👍

**Review by nickrum — CHANGES_REQUESTED** [review]
Instead of creating a new member variable on the flow manager, we should be able to pass this option to the controller through the flow's handler like this:

```ts
const handler = async (data: unknown, context: Record<string, unknown>) => {
	const cacheEnabled = flow.options['cacheEnabled'];

	if (flow.options['async']) {
		this.executeFlow(flow, data, context);
		return { result: undefined, cacheEnabled };
	} else {
		return { result: await this.executeFlow(flow, data, context), cacheEnabled };
	}
};
```

**Review by licitdev — COMMENTED** [review]
@nickrum Great suggestion! Simplifying the patch. Thanks! 🙌

**Review by nickrum — APPROVED** [review]

File: .changeset/modern-eels-judge.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+"@directus/app": minor
+"@directus/api": minor
+"tests-blackbox": minor
+---
+
+Added `Cache` option so that caching can be disabled for GET requests to Flows webhook trigger.
```
Comment (paescuj) [inline .changeset/modern-eels-judge.md:7]: The majority of the existing changesets don't end with a period:

```suggestion
Added `Cache` option so that caching can be disabled for GET requests to Flows webhook trigger
```

**Review by paescuj — APPROVED** [review]
Works great! 😃
